<launch>
  <arg name="task" default="freemotion" />

  <group if="$(eval task == 'freemotion')">
    <rosparam command="load" file="$(find cbf_tanks_pkg)/config/var_stiff_task_param.yaml"/>
    <param name="tank_type" type="string" value="$(arg tank_type)" allowed="first|firstcbf|cbf|none"/>
    <node name="cbftank_main" pkg="cbf_tanks_pkg" type="cbftank_main" output="screen" />
  </group>

  <group if="$(eval task == 'force')">
    <rosparam command="load" file="$(find cbf_tanks_pkg)/config/force_task_param.yaml"/>
    <param name="tank_type" type="string" value="$(arg tank_type)" allowed="cbf|none"/>
    <node name="force_control_node" pkg="cbf_tanks_pkg" type="force_control_node" output="screen" />
  </group>

  <group if="$(eval task == 'KE')">
    <rosparam command="load" file="$(find cbf_tanks_pkg)/config/KE_task_param.yaml"/>
    <param name="tank_type" type="string" value="$(arg tank_type)" allowed="cbf|none"/>
    <node name="KE_limitation_node" pkg="cbf_tanks_pkg" type="KE_limitation_node" output="screen" />
  </group>
</launch>
